crates/bitwarden-core/src/key_management/master_password.rs

```diff
@@ -74,6 +109,43 @@ fn kdf_parse_nonzero_u32(value: impl TryInto<u32>) -> Result<NonZeroU32, MasterP
         .ok_or(MasterPasswordError::KdfMalformed)
 }
 
+/// Represents the data required to unlock with the master password.
+#[allow(missing_docs)]
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+#[cfg_attr(feature = "uniffi", derive(uniffi::Record))]
+#[cfg_attr(
+    feature = "wasm",
+    derive(tsify::Tsify),
+    tsify(into_wasm_abi, from_wasm_abi)
+)]
+pub struct MasterPasswordAuthenticationData {
+    pub kdf: Kdf,
+    pub salt: String,
+    pub master_password_authentication_hash: B64,
+}
+
+impl MasterPasswordAuthenticationData {
+    pub(crate) fn derive(
+        password: &str,
+        kdf: &Kdf,
+        salt: &str,
+    ) -> Result<Self, MasterPasswordError> {
+        let master_key = MasterKey::derive(password, salt, kdf)
+            .map_err(|_| MasterPasswordError::InvalidKdfConfiguration)?;
+        let hash = master_key.derive_master_key_hash(
+            password.as_bytes(),
+            bitwarden_crypto::HashPurpose::ServerAuthorization,
+        );
+
+        Ok(Self {
+            kdf: kdf.clone(),
```

Comment:
(nit) Since it's a reference, we should use `to_owned()` instead of `clone()`
```suggestion
            kdf: kdf.to_owned(),
```